x-pack/packages/kbn-elastic-assistant-common/impl/utils/bedrock.test.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EventStreamCodec } from '@smithy/eventstream-codec';
+import { fromUtf8, toUtf8 } from '@smithy/util-utf8';
+import { handleBedrockChunk } from '../..';
+
+const getContentBlockDelta = (completion: string) => ({
+  type: 'content_block_delta',
+  index: 0,
+  delta: { type: 'text_delta', text: completion },
+});
+
+const messageStart = {
+  type: 'message_start',
+  message: {
+    id: 'msg_01Mp3df6ic3BZPJqJGS23qsx',
+    type: 'message',
+    role: 'assistant',
+    content: [],
+    model: 'claude-3-sonnet-28k-20240229',
+    stop_reason: null,
+    stop_sequence: null,
+    usage: { input_tokens: 10, output_tokens: 1 },
+  },
+};
+const contentBlockStart = {
+  type: 'content_block_start',
+  index: 0,
+  content_block: { type: 'text', text: '' },
+};
+
+const contentBlockStop = { type: 'content_block_stop', index: 0 };
+const messageDelta = {
+  type: 'message_delta',
+  delta: { stop_reason: 'end_turn', stop_sequence: null },
+  usage: { output_tokens: 5 },
+};
+const messageStop = {
+  type: 'message_stop',
+  'amazon-bedrock-invocationMetrics': {
+    inputTokenCount: 10,
+    outputTokenCount: 5,
+    invocationLatency: 629,
+    firstByteLatency: 515,
+  },
+};
+
+const mockChunks = [
+  messageStart,
+  contentBlockStart,
+  getContentBlockDelta('My'),
+  getContentBlockDelta(' new'),
+  getContentBlockDelta(' message'),
+  contentBlockStop,
+  messageDelta,
+  messageStop,
+].map((c) => encodeBedrockResponse(c));
```

Comment:
@pmuellr the reason I left the type loose is because the stream chunks can be so many types of objects. Bedrock does not document this well. I followed the [LangChain Bedrock implementation](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-community/src/utils/bedrock.ts#L255-L307) to parse the data, but something could have been missed so our parser will default to empty string

I will add in types for the known fields, but I think I should keep the type loose so we don't error when something unexpected happens. What do you think?

Reply:
Here we go: https://github.com/elastic/kibana/pull/179304/commits/2ae2605d1d68e8dfed55d14ae76ce91b180df280